packages/aws-cdk/lib/api/aws-auth/awscli-compatible.ts

```diff
@@ -182,7 +182,10 @@ function caBundlePathFromEnvironment(): string | undefined {
 function shouldPrioritizeEnv() {
   const id = process.env.AWS_ACCESS_KEY_ID || process.env.AMAZON_ACCESS_KEY_ID;
   const key = process.env.AWS_SECRET_ACCESS_KEY || process.env.AMAZON_SECRET_ACCESS_KEY;
-  process.env.AWS_SESSION_TOKEN = process.env.AWS_SESSION_TOKEN || process.env.AMAZON_SESSION_TOKEN;
+  const sessionToken = process.env.AWS_SESSION_TOKEN || process.env.AMAZON_SESSION_TOKEN;
+  if (sessionToken) {
+    process.env.AWS_SESSION_TOKEN = sessionToken;
+  }
```

Comment:
Why don't we do this inside the `if` that checks for `id` and `key`?

Why do we need to do this at all, anyway? Is it because `AMAZON_ACCESS_KEY_ID` is no longer supported, and it must be `AWS_ACCESS_KEY_ID`?